app/scripts/modules/amazon/src/loadBalancer/AmazonLoadBalancersTag.tsx

```diff
@@ -130,26 +119,32 @@ export class AmazonLoadBalancersTag extends React.Component<ILoadBalancersTagPro
   };
 
   public componentDidMount(): void {
-    this.mounted = true;
     this.loadBalancersRefreshUnsubscribe = this.props.application.getDataSource('loadBalancers').onRefresh(null, () => {
       this.forceUpdate();
     });
+
+    LoadBalancerDataUtils.populateLoadBalancers(this.props.application, this.props.serverGroup).then(loadBalancers => {
+      this.setState({ loadBalancers, isLoading: false });
```

Comment:
I think you still want the `if (this.mounted)` check here - if the user navigates away from the page before the callback is executed, the `setState` call here throws a warning or error.

(same as below)

Reply:
I believe that would have been correct since we were loading it in the constructor. But now the logic is moved to componentDidMount, I think it should not matter. I will thoroughly test it and confirm.

Reply:
The problem is not calling it when the component mounts, it's that we are sometimes calling `setState` after the component has _unmounted_, due to the callback.

Reply:
Yes! I verified that thats true. Reverted to using mounted.